kavdi-brian/scripts/articleView.js

```diff
@@ -0,0 +1,154 @@
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = function() {
+  $('.main-nav').on('click', '.tab', function() {
+    $('.tab-content').hide();
+    $('#' + $(this).data('content')).fadeIn();
+  });
+
+  $('.main-nav .tab:first').click();
+};
+
+articleView.setTeasers = function() {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
+      $(this).html('Show Less &larr;');
+    } else {
+      $('body').animate({
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+articleView.initNewArticlePage = function() {
+  // DONE: Ensure the main .tab-content area is revealed. We might add more tabs later or otherwise edit the tab navigation.
+  $('.tab-content').show();
+
+  // DONE: The new articles we create will be copy/pasted into our source data file.
+  // We need to allow the container to generate the JSON that will be copy-pasted.
+  // We can hide it for now, and show it once we have data to export.
+  $('#article-export').hide();
+
+  $('#article-json').on('focus', function(){
+    this.select();
+  });
+
+  // DONE: Add an event handler to...
+  //  - update the preview field if any inputs change -- DONE
+  //  - update the export field if any inputs change.
+  $('form').on('change', function(event){
+    /*
+    - Grab the template's HTML
+    - Compile the template into a callable function
+    - Call the function with the data with which you want to fill the template
+    - Append the filled template to the DOM
+    */
+    var sourceHTML = $('#preview-template').html();
+    var compiledTemplate = Handlebars.compile(sourceHTML);
+
+    // if the published box is checked, then calculate how many days ago this article was published, and put that in my article preview.
+    // if the published box is not checked, then just put the text "(draft)" in my article preview
+
+    var today = new Date();
+    var dateText = today.toLocaleDateString(); // this is for the JSON text later. Figure out where to put it
+
+    var formData = {
+      title: this.title.value,
+      body: this.body.value,
+      author: this.author.value,
+      url: this.url.value,
+      publishStatus: this.publishStatus.checked ? `published 0 days ago` : '(draft)',
+      category: this.category.value
+    };
```

Comment:
Just so you know, if you instantiate a new `Article` object using the data from the form instead of just leaving it as an object literal, you can make use of the methods that exist on `Article` like `.toHtml()`.